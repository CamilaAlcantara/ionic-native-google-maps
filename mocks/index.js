module.exports = {
  BaseClass: require('./BaseClass'),
  BaseArrayClass: require('./BaseArrayClass'),
};