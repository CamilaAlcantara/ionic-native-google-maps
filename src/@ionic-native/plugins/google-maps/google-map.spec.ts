import { GoogleMap, GoogleMaps } from '../../../../dist/@ionic-native/plugins/google-maps';

import { GoogleMapMock, GoogleMapsMock } from '../../../test/mocks';
import { mockCordova, nextId } from '../../../test/utils';


describe('GoogleMap', () => {
  let googleMaps: GoogleMapsMock;
  let googleMap: GoogleMapMock;

  beforeEach(() => {
    googleMaps = new GoogleMapsMock();
    googleMap = new GoogleMapMock();

    mockCordova({
      GoogleMaps: googleMaps,
      Map: googleMap,
    });
  });

  describe('should throw', () => {
    it('when the element does not exist', async () => {
      const _ = new GoogleMap(nextId());
      const [promise] = googleMap.getMap.mock.calls[0];
      await expect(promise).rejects.toMatchSnapshot();
    });

    it('when the element is too small', async () => {
      const mapId = nextId();

      document.body.innerHTML = `
        <div class="show-page">
          <div id="${mapId}" style="width: 99px; height: 99px;"></div>
        </div>
      `;

      const _ = new GoogleMap(mapId);
      const [promise] = googleMap.getMap.mock.calls[0];
      expect(googleMap.getMap).toHaveBeenCalled();
      await expect(promise).rejects.toMatchSnapshot();
    });
  });

  it('should work...', async () => {
    const mapId = nextId();

    document.body.innerHTML = `
      <div class="show-page">
        <div id="${mapId}" style="width: 100px; height: 100px;"></div>
      </div>
    `;

    console.log(document.querySelector(`.show-page #${mapId}`));

    const _ = new GoogleMap(mapId);
    const [promise] = googleMap.getMap.mock.calls[0];
    expect(googleMap.getMap).toHaveBeenCalled();
    await expect(promise).resolves.toMatchSnapshot();
  });
});
