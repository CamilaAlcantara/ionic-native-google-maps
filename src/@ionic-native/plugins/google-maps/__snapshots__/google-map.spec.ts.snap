// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap with cordova should throw when the element does not exist 1`] = `[Error: Can not find the element [#uniqueId-0]]`;

exports[`GoogleMap with cordova should throw when the element is too small 1`] = `[Error: DIV[#uniqueId-1] is too small. Must be bigger than 100x100.]`;

exports[`GoogleMap with cordova should work when the dom element is present and an ID is passed 1`] = `
Array [
  <div
    id="uniqueId-2"
    style="width: 100px; height: 100px;"
  />,
  undefined,
]
`;

exports[`GoogleMap with cordova should work when the dom element is present and an element is passed 1`] = `
<div
  id="uniqueId-3"
  style="width: 100px; height: 100px;"
/>
`;

exports[`GoogleMap without cordova should display a message in the target element 1`] = `
"
        <div class=\\"show-page\\">
          <div id=\\"uniqueId-4\\" style=\\"width: 100px; height: 100px; position: relative;\\"><div style=\\"color: red; position: absolute; width: 80%; height: 50%; top: 0px; bottom: 0px; right: 0px; left: 0px; padding: 0px; margin: auto;\\">[GoogleMaps]<br>You need to execute \\"$&gt; ionic cordova run browser\\".<br>\\"$&gt; ionic serve\\" is not supported.</div></div>
        </div>
      "
`;

exports[`GoogleMap without plugin should display a message in the target element 1`] = `
"
        <div class=\\"show-page\\">
          <div id=\\"uniqueId-5\\" style=\\"width: 100px; height: 100px; position: relative;\\"><div style=\\"color: red; position: absolute; width: 80%; height: 50%; top: 0px; bottom: 0px; right: 0px; left: 0px; padding: 0px; margin: auto;\\">[GoogleMaps]<br>cordova-plugin-googlemaps is not installed or not ready yet.</div></div>
        </div>
      "
`;
